### DISCUSSION TITLE
Running in single precision

### DISCUSSION THREAD

**Discussion by jmcnrc** [Nek5000/nekRS#409] [Q&A] [answered]
Hi,

I'm doing some CUDA testing with a non-compute-focused GPU (so the FP64:FP32 ratio is 1:64 rather than the 1:2 of V100, A100, Tesla, etc.).  I have about 1/20th the FP64 power of an A100, but 1/2 to 1/3 the memory bandwidth, so I suspect there is a significant compute bottleneck.  My own testing agrees with that discussed in _NekRS, a GPU-Accelerated Spectral Element Navier-Stokes Solver_ that there is not much of a speedup when using single precision for the coarse grid solve.

However, I assume most of the compute work is **not** for the coarse grid solve (?), and that by using FP32 for both the coarse grid solve and everywhere else, my compute bottleneck and would be greatly reduced.  I would also expect to get more performance for the memory bandwidth, since the word size would be smaller.  I obviously wouldn't expect to run 64x faster, but I would find it very interesting if I can shift the bottleneck from compute to memory bandwidth.

I also do realize that there can be numerical issues with going to FP32, but I think it would be interesting to try out.

TL;DR
Is there a way to run everything in single precision with CUDA?  If so, how?

Thanks,
Justin

**Comment by stgeke** [comment, marked as answer]
That's not supported at the moment.

**Reply by jmcnrc** [reply]
Thanks! Any plans for it in the future?  I assume it would be low priority, as most users will run double precision.

**Reply by stgeke** [reply]
It's in our backlog but no plans to add it any time soon.

**Reply by jmcnrc** [reply]
That makes sense, and sounds good.  Thanks again!
